### PR TITLE
Fix for "calculated and received parameters differ in parameter" in suffix bootstrapping

### LINKED ISSUE #1059: "Calculated and received parameters differ in parameter" for suffix bootstrapping
Example from log: 

00:43:00.225 WARN  [ctor.default-dispatcher-9] o.e.n.state.DigestState - Invalid block 390c62759e5c93395b8de35115942b64ca7b8b2540613d2399fef67861f25a2b, reason: scorex.core.validation.MalformedModifierError: At the beginning of the epoch, the extension should contain all the system parameters. Calculated and received parameters differ in parameter 4 (1093684 != 1000000)
00:43:09.869 WARN  [ctor.default-dispatcher-4] o.e.n.ErgoNodeViewSynchronizer - Modifier 6eea0ffab06affaffa7b2c0a1f48de629f752c5bff6f54f802b2d83f6b20702a is permanently invalid
scorex.core.validation.MalformedModifierError: A header for the block section should not be marked as invalid. header=d0c7481d12b05a3a757bb3e997d3a08930bcb7c8002464cd2ea70c9cf8b0e2ec, id=6eea0ffab06affaffa7b2c0a1f48de629f752c5bff6f54f802b2d83f6b20702a
	at scorex.core.validation.ModifierValidator$.fatal(ModifierValidator.scala:44)
	at org.ergoplatform.settings.ValidationRules$.fatal(ValidationRules.scala:298)
	at org.ergoplatform.settings.ValidationRules$.$anonfun$rulesSpec$40(ValidationRules.scala:145)
	at org.ergoplatform.settings.ErgoValidationSettings.$anonfun$getError$1(ErgoValidationSettings.scala:33)
	at scala.Option.map(Option.scala:230)
	at org.ergoplatform.settings.ErgoValidationSettings.getError(ErgoValidationSettings.scala:33)
	at scorex.core.validation.ValidationState.$anonfun$validate$1(ModifierValidator.scala:93)
	at scorex.core.validation.ValidationState.pass(ModifierValidator.scala:72)
	at scorex.core.validation.ValidationState.validate(ModifierValidator.scala:93)
	at scorex.core.validation.ValidationState.validateNot(ModifierValidator.scala:98)
	at scorex.core.validation.ValidationState.validateSemantics(ModifierValidator.scala:126)
	at org.ergoplatform.nodeView.history.storage.modifierprocessors.FullBlockSectionProcessor$PayloadValidator.validate(FullBlockSectionProcessor.scala:98)
	at org.ergoplatform.nodeView.history.storage.modifierprocessors.FullBlockSectionProcessor.$anonfun$validate$1(FullBlockSectionProcessor.scala:47)
	at scala.Option.map(Option.scala:230)
	at org.ergoplatform.nodeView.history.storage.modifierprocessors.FullBlockSectionProcessor.validate(FullBlockSectionProcessor.scala:45)
	at org.ergoplatform.nodeView.history.storage.modifierprocessors.FullBlockSectionProcessor.validate$(FullBlockSectionProcessor.scala:44)
	at org.ergoplatform.nodeView.history.ErgoHistory$$anon$3.validate(ErgoHistory.scala:250)
	at org.ergoplatform.nodeView.history.ErgoHistoryReader.applicableTry(ErgoHistoryReader.scala:216)
	at org.ergoplatform.nodeView.history.ErgoHistoryReader.applicableTry$(ErgoHistoryReader.scala:211)
	at org.ergoplatform.nodeView.history.ErgoHistory$$anon$3.applicableTry(ErgoHistory.scala:250)
	at org.ergoplatform.nodeView.history.ErgoHistory$$anon$3.applicableTry(ErgoHistory.scala:250)
	at scorex.core.network.NodeViewSynchronizer.scorex$core$network$NodeViewSynchronizer$$validateAndSetStatus(NodeViewSynchronizer.scala:296)
	at scorex.core.network.NodeViewSynchronizer$$anonfun$modifiersFromRemote$1.$anonfun$applyOrElse$12(NodeViewSynchronizer.scala:281)
	at scorex.core.network.NodeViewSynchronizer$$anonfun$modifiersFromRemote$1.$anonfun$applyOrElse$12$adapted(NodeViewSynchronizer.scala:281)

### PR DESCRIPTION
**Pull request by kushti**
Closes #1059

### CODE REVIEW

File: src/main/scala/org/ergoplatform/nodeView/state/ErgoStateContext.scala [hunk 1]
```diff
@@ -129,28 +128,42 @@ class ErgoStateContext(val lastHeaders: Seq[Header],
...
+              }
+
+              currentValidationState
+                .validate(exBlockVersion, calculatedParams.blockVersion == header.version, s"${calculatedParams.blockVersion} == ${header.version}")
+                .validateNoFailure(exMatchParameters, Parameters.matchParameters(parsedParams, calculatedParams))
+                .validate(exMatchValidationSettings, parsedSettings == calculatedSettings, s"$parsedSettings vs $calculatedSettings")
+          }.result
+      }.result
+      .toTry
+      .flatMap(_ => parsedParamsTry.flatMap(p => parsedValidationSettingsTry.map(vs => (p, vs))))
```
Comment (aslesarenko) [inline src/main/scala/org/ergoplatform/nodeView/state/ErgoStateContext.scala]: Why parameter of flatMap is ignored?

Comment (kushti) [reply src/main/scala/org/ergoplatform/nodeView/state/ErgoStateContext.scala]: @aslesarenko it is of Unit type

File: src/main/scala/org/ergoplatform/nodeView/state/ErgoStateContext.scala [hunk 2]
```diff
@@ -129,28 +128,42 @@ class ErgoStateContext(val lastHeaders: Seq[Header],
...
-      .flatMap(_ => parsedParamsTry.flatMap(p => parsedValidationTry.map(vs => (p, vs))))
-
+        case (vs, parsedParams) =>
+          vs.validateTry(parsedValidationSettingsTry, e => ModifierValidator.fatal("Failed to parse validation settings", e)) {
+            case (currentValidationState, parsedSettings) =>
+
+              /*
+               Calculating params for parameters and validation rules based on local blockchain state, to compare them
+               with parsed ones then.
+               If current parameters height is equal to 0, it means that whether it is first epoch after genesis,
```
[the remark this answers is not in the document]
Comment (kushti) [reply src/main/scala/org/ergoplatform/nodeView/state/ErgoStateContext.scala]: @aslesarenko whether is valid here

File: src/main/scala/org/ergoplatform/nodeView/state/ErgoStateContext.scala [hunk 3]
```diff
@@ -129,28 +128,42 @@ class ErgoStateContext(val lastHeaders: Seq[Header],
...
-      }.result.toTry
-      .flatMap(_ => parsedParamsTry.flatMap(p => parsedValidationTry.map(vs => (p, vs))))
-
+        case (vs, parsedParams) =>
+          vs.validateTry(parsedValidationSettingsTry, e => ModifierValidator.fatal("Failed to parse validation settings", e)) {
+            case (currentValidationState, parsedSettings) =>
+
+              /*
+               Calculating params for parameters and validation rules based on local blockchain state, to compare them
+               with parsed ones then.
```
[the remark this answers is not in the document]
Comment (kushti) [reply src/main/scala/org/ergoplatform/nodeView/state/ErgoStateContext.scala]: done

**Pull request merged**